### DISCUSSION TITLE
Is there a way to create a hook during file renaming/movement?

### DISCUSSION THREAD

**Discussion by zhou13** [mikavilpas/yazi.nvim#493] [General]
I am using https://github.com/alexpasmantier/pymple.nvim and I am interested in migrating to yazi.nvim. `pymple.nvim` requires a a hook on file renaming to setup: https://github.com/alexpasmantier/pymple.nvim/blob/main/lua/pymple/hooks.lua#L53-L71 according to https://github.com/alexpasmantier/pymple.nvim/issues/46. Is there a way to do it with yazi.nvim?

**Comment by mikavilpas** [comment]
Currently it's not possible, but I can add support for this. I think adding a custom event is the right way to do it.

When support for an event like this has been added, you should be able to add custom support for pymple right away.

Later on, I can suggest pymple to adopt yazi.nvim integration along with the others that it has.

Finally, any custom integration can be removed.

Would you like to create an issue for this?

**Reply by zhou13** [reply]
Added: https://github.com/mikavilpas/yazi.nvim/issues/494

**Reply by mikavilpas** [reply]
Ok now it should be done. You can add an event handler like this in your config
to try it out:

```lua
vim.api.nvim_create_autocmd("User", {
  pattern = "YaziRenamedOrMoved",
  -- see `:help event-args`
  ---@param event {data: YaziNeovimEvent.YaziRenamedOrMovedData}
  callback = function(event)
    -- printing the messages will allow seeing them with `:messages`
    print(vim.inspect({ "Just received an event!", event.data }))
  end,
})
```

**Comment by mikavilpas** [comment]
Looks like this is also discussed ~here https://github.com/mikavilpas/yazi.nvim/discussions/493#discussioncomment-10846895~

edit: I meant here https://github.com/alexpasmantier/pymple.nvim/issues/46

**Reply by alexpasmantier** [reply]
Hi, thanks for adding the feature.
Just tried it and it seems I can't detect any events when renaming or moving files with Yazi, even when setting up your example snippet.
Any idea why that might be?

I'm not sure the logs are of any help here but pasting them just in case:
```
  2024-10-04T21:07:49.630258Z  WARN yazi_adapter::adapter: [Adapter] Could not identify XDG_SESSION_TYPE
    at yazi-adapter/src/adapter.rs:108

  2024-10-04T21:07:49.630291Z  WARN yazi_adapter::adapter: [Adapter] Falling back to chafa
    at yazi-adapter/src/adapter.rs:120

  2024-10-04T21:07:49.631212Z  WARN yazi_adapter::ueberzug: ueberzugpp spawning failed: No such file or directory (os error 2)
    at yazi-adapter/src/ueberzug.rs:95

  2024-10-04T21:08:39.085508Z  WARN yazi_adapter::adapter: [Adapter] Could not identify XDG_SESSION_TYPE
    at yazi-adapter/src/adapter.rs:108

  2024-10-04T21:08:39.085530Z  WARN yazi_adapter::adapter: [Adapter] Falling back to chafa
    at yazi-adapter/src/adapter.rs:120

  2024-10-04T21:08:39.086419Z  WARN yazi_adapter::ueberzug: ueberzugpp spawning failed: No such file or directory (os error 2)
    at yazi-adapter/src/ueberzug.rs:95

```

**Reply by mikavilpas** [reply]
Hmm I will look into this a bit. In the meantime, could you follow this and submit yazi.nvim logs as well as the checkhealth output?

https://github.com/mikavilpas/yazi.nvim/blob/main/documentation/reproducing-issues.md#logging

**Reply by alexpasmantier** [reply]
Thanks, your fix worked like a charm 🙏🏻
